### PR TITLE
Support Script Sort

### LINKED ISSUE #133: Support Script Sort

### PR DESCRIPTION
**Pull request by dbulaja98**
Resolves #133

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/ElasticSort.scala [hunk 1]
```diff
@@ -19,8 +19,8 @@ package zio.elasticsearch
 import zio.elasticsearch.query.sort._
 
 object ElasticSort {
-  def sortBy[S](field: Field[S, _]): Sort =
-    SortOptions(
+  def sortByField[S](field: Field[S, _]): SortByField =
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticSort.scala:22]: Let's name it `sortBy`.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticSort.scala [hunk 2]
```diff
@@ -40,4 +40,14 @@ object ElasticSort {
       order = None,
       unmappedType = None
     )
+
+  def sortByScript(script: String, sourceType: SourceType): SortByScript =
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticSort.scala:44]: I would create `Script` as `AnyVal` or `Newtype`, and use `sortBy` here as well. 

```
sortBy(field: String, ...)
sortBy(script: Script, ...)
```

File: modules/library/src/main/scala/zio/elasticsearch/query/sort/Sort.scala [hunk 3]
```diff
@@ -68,6 +73,66 @@ private[elasticsearch] final case class SortOptions(
...
+        ).collect { case Some(obj) => obj }: _*
+      )
+    )
+
+  private def scriptToJson: Json =
+    Obj(
+      List(
+        self.lang.map(lang => "lang" -> lang.toJson),
+        Some("source" -> source.toJson),
+        if (params.nonEmpty)
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/query/sort/Sort.scala:121]: Let's add parentheses.

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

File: modules/library/src/main/scala/zio/elasticsearch/query/sort/Sort.scala [hunk 4]
```diff
@@ -68,6 +73,66 @@ private[elasticsearch] final case class SortOptions(
...
+
+  def paramsToJson: Json =
+    Obj(
+      "_script" -> Obj(
+        List(
+          Some("type"   -> self.sourceType.toString.toJson),
+          Some("script" -> scriptToJson),
+          self.order.map(order => "order" -> order.toString.toJson),
+          self.mode.map(mode => "mode" -> mode.toString.toJson)
+        ).collect { case Some(obj) => obj }: _*
```
Comment (arnoldlacko) [inline modules/library/src/main/scala/zio/elasticsearch/query/sort/Sort.scala:112]: I think this `collect` could be `flatten`

**Review by drmarjanovic — APPROVED** [review]

**Pull request merged**